src/main/java/net/mcreator/workspace/Workspace.java

```diff
@@ -496,4 +497,22 @@ public boolean isRegenerateRequired() {
 		this.workspaceSettings.setWorkspace(this);
 	}
 
+	public static boolean isWorkspaceLockedToVersion(File workspaceFile) {
+		String workspaceString = FileIO.readFileToString(workspaceFile);
+		JsonObject workspaceJson = WorkspaceFileManager.gson.fromJson(workspaceString, JsonObject.class);
+		if (workspaceJson.has("workspaceSettings") && workspaceJson.has("mcreatorVersion")) {
+			JsonObject workspaceSettings = workspaceJson.getAsJsonObject("workspaceSettings");
+			return workspaceSettings.has("lockWorkspaceToCurrentVersion") && workspaceSettings.get(
+					"lockWorkspaceToCurrentVersion").getAsBoolean();
+		}
+		return false;
+	}
+
+	public static boolean workspaceVersionMatchesCurrentVersion(File workspaceFile) {
+		String workspaceString = FileIO.readFileToString(workspaceFile);
+		JsonObject workspaceJson = WorkspaceFileManager.gson.fromJson(workspaceString, JsonObject.class);
+		Long workspaceVersion = workspaceJson.get("mcreatorVersion").getAsLong();
+		return workspaceVersion.equals(Launcher.version.versionlong);

```

Comment:
Long should be compared by == and not equals. This method should return long and be getter for workspace version so it is more universal.